openfeature/telemetry/telemetry.go

```diff
@@ -6,90 +7,99 @@ import (
 	"github.com/open-feature/go-sdk/openfeature"
 )
 
+// EvaluationEvent represents an event that is emitted when a flag is evaluated.
+// It is intended to be used to record flag evaluation events as OpenTelemetry log records.
+// See the OpenFeature specification [Appendix D: Observability] and
+// the OpenTelemetry [Semantic conventions for feature flags in logs] for more information.
+//
+// [Appendix D: Observability]: https://openfeature.dev/specification/appendix-d
+// [Semantic conventions for feature flags in logs]: https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/
 type EvaluationEvent struct {
-	Name       string
+	// Name is the name of the event.
+	// It is always "feature_flag.evaluation".
+	Name string
+	// Attributes represents the event's attributes.
 	Attributes map[string]any
-	Body       map[string]any
+	// Body is the flag's value and is only present if the flag's variant is empty.
+	Body map[string]any
 }
 
+// The OpenTelemetry compliant event attributes for flag evaluation.
 const (
-	// The OpenTelemetry compliant event attributes for flag evaluation.
-	// Specification: https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/
-
-	TelemetryKey       string = "feature_flag.key"
-	TelemetryErrorCode string = "error.type"
-	TelemetryVariant   string = "feature_flag.variant"
-	TelemetryContextID string = "feature_flag.context.id"
-	TelemetryErrorMsg  string = "feature_flag.evaluation.error.message"
-	TelemetryReason    string = "feature_flag.evaluation.reason"
-	TelemetryProvider  string = "feature_flag.provider_name"
-	TelemetryFlagSetID string = "feature_flag.set.id"
-	TelemetryVersion   string = "feature_flag.version"
-
-	// Well-known flag metadata attributes for telemetry events.
-	// Specification: https://openfeature.dev/specification/appendix-d#flag-metadata
-	TelemetryFlagMetaContextId string = "contextId"
-	TelemetryFlagMetaFlagSetId string = "flagSetId"
-	TelemetryFlagMetaVersion   string = "version"
-
-	// OpenTelemetry event body.
-	// Specification: https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/
-	TelemetryBody string = "value"
```

Comment:
I removed the word `Telemetry` from const names because the package name is `telemetry`.